app/models/ability.rb

```diff
@@ -8,6 +8,7 @@ def initialize(user)
      when "User"
       can [:read, :filter], Receipt, user: user
      when "Retailer"
+      can [:read, :filter], Voucher, retailer: user
```

Comment:
this is the Ability, when someone is logged in on the website. 

If it is a User it can only index, show and filter the receipts

When it is a retailer it can only do that with voucher

as it is now, everything can be modifyied :)